R/get_data.R

```diff
@@ -59,47 +56,84 @@ bcdc_get_data.bcdc_record <- function(record, format = NULL, resource = NULL, ..
 }
 
 #' @export
-bcdc_get_data.character <- function(record, format = NULL, resource = NULL, ...) {
+bcdc_get_data.character <- function(record, resource = NULL, ...) {
```

Comment:
This function is getting long and a bit hard to follow the logic (though I recognize that with some many combinations of factors it's bound to be a bit convoluted). I wonder if we can wrap up some discrete bits into separate functions and/or move some of it up into `bcdc_get_resource()` so we're only parsing the resource once?

Reply:
Absolutely agree. Is `bcdc_get_resource()` a function you are envisioning? I think that we could move all the menu stuff into a function or two. The "download -> read in data" code could also be moved into its own function. Lastly the parsing of resources into a dataframe can also be moved into a function. That would keep the logic in main body but all the worker bee function abstracted away. Can you think of anywhere else?

Reply:
Agreed with those areas to abstract out. Also, I think I meant move up into `bcdc_get_record()` - does that make sense?

Reply:
I think my last few commits resolve this, and close #71, at least well enough - what do you think @boshek?